src/guidellm/benchmark/entrypoints.py

```diff
@@ -195,45 +216,74 @@ async def resolve_request_loader(
     :param console: Console instance for progress reporting, or None
     :param dataloader_kwargs: Additional arguments passed to DataLoader initialization
     :return: Configured DataLoader instance for GenerationRequest objects
+    :raises ValueError: If request formatter type is not registered in
+        PreprocessorRegistry
+    :raises TypeError: If registered request formatter is not a RequestFormatter
+        subclass
     """
     console_step = (
         console.print_update_step(title=f"Initializing request loader from {data}")
         if console
         else None
     )
 
-    if not isinstance(data_column_mapper, DatasetPreprocessor):
+    data_column_mapper_instance: DatasetPreprocessor
+    if isinstance(data_column_mapper, DatasetPreprocessor):
+        data_column_mapper_instance = data_column_mapper
+    else:
         column_mappings = (
             data_column_mapper if isinstance(data_column_mapper, dict) else None
         )
-        data_column_mapper = GenerativeColumnMapper(
-            column_mappings=column_mappings,
+        data_column_mapper_instance = GenerativeColumnMapper(
+            column_mappings=column_mappings  # type: ignore[arg-type]
         )
-    if not isinstance(data_request_formatter, DatasetPreprocessor):
-        request_type = (
-            data_request_formatter
-            if isinstance(data_request_formatter, str)
-            else data_request_formatter.pop("request_type", "chat_completions")
-        )
-        data_request_formatter = PreprocessorRegistry.get_registered_object(
-            request_type
-        )(
+
+    data_request_formatter_instance: RequestFormatter
+    if isinstance(data_request_formatter, RequestFormatter):
+        data_request_formatter_instance = data_request_formatter
+    else:
+        if isinstance(data_request_formatter, str):
+            request_type = data_request_formatter
+            formatter_kwargs: dict[str, Any] = {}
+        else:
+            # Extract request_type from formatter dictionary
+            formatter_dict = dict(data_request_formatter)
+            request_type = formatter_dict.pop("request_type", "chat_completions")
```

Comment:
```suggestion
            request_type = formatter_dict.pop("request_type", settings.preferred_route)
```

Also need to update `preferred_route` to be `chat_completions`.